src/app/hopefuelidlist/page.js

```diff
@@ -16,6 +24,25 @@ const HopeFuelIdListPage = () => {
     );
   }, [searchText]);
 
+  const handleScroll = () => {
+    if (
+      window.innerHeight + window.scrollY >= document.body.offsetHeight - 100 &&
+      visibleCount < filteredData.length &&
+      !loading
+    ) {
+      setLoading(true);
+      setTimeout(() => {
+        setVisibleCount((prev) => prev + 10);
+        setLoading(false);
+      }, 1000);
+    }
+  };
+
+  useEffect(() => {
+    window.addEventListener("scroll", handleScroll);
+    return () => window.removeEventListener("scroll", handleScroll);
+  }, [visibleCount, filteredData.length, loading]);
```

Comment:
This dependency array could be a source of error.

တကယ်လို့ loading useState တွေ ခဏခဏ ပြောင်းနေရင်  Event listener တွေ ခဏခဏ ထည့်မိသလို ဖြစ်နေတယ်

visuable count and filteredData ကို လည်း တူတူဘဲ 

```suggestion
  }, []);
```

ဒီလိုလုပ်လိုက်ရင် တခါဘဲ​ခေါ်တော့မှာ

Reply:
> This dependency array could be a source of error.
> 
> တကယ်လို့ loading useState တွေ ခဏခဏ ပြောင်းနေရင် Event listener တွေ ခဏခဏ ထည့်မိသလို ဖြစ်နေတယ်
> 
> visuable count and filteredData ကို လည်း တူတူဘဲ
> 
> ဒီလိုလုပ်လိုက်ရင် တခါဘဲ​ခေါ်တော့မှာ

Thanks Ko Dee. I will resolve this review.